File: PROGRAMMERS/LEVEL1/[1차] 비밀지도.cpp
#include <string>
#include <vector>

using namespace std;


vector<string> solution(int n, vector<int> arr1, vector<int> arr2) {
    vector<string> answer;
    string st;
    for (int i = 0; i < n; i++) {
        int num = arr1[i] | arr2[i];
        st = "";
        for (int j = 0; j < n; j++) {
            if (num % 2 == 0) st = " " + st;
            else st = "#" + st;
            num /= 2;
        }
        answer.push_back(st);
    }

    return answer;
}

File: PROGRAMMERS/LEVEL1/[1차] 비밀지도.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

class Solution {
    public List solution(int n, int[] arr1, int[] arr2) {
        List<String> answer = new ArrayList<String>();
        
        for(int i=0;i<n;i++) {
        	int num = arr1[i] | arr2[i];
        	String str="";
        	
        	for(int j=0;j<n;j++) {
        		if(num%2==0) str=" "+str;
        		else str="#"+str;
        		
        		num/=2;
        	}
        	answer.add(str);
        }
        
        return answer;
    }
}